### ISSUE TITLE
Flask's _request_ctx_stack is deprecated

### ISSUE THREAD

**Issue by ivanovmg** [pytest-dev/pytest-flask#164]
**Describe the bug**
Flask's ``_request_ctx_stack`` is deprecated and will be removed in Flask 2.4.
Currently ``pytest-flask`` relies on this function in https://github.com/pytest-dev/pytest-flask/blob/04c12559451d6c9da0633d604f094e0c9a0e6bb2/pytest_flask/fixtures.py#L96

**To Reproduce**
``` bash
$ pip list | grep pytest-flask
pytest-flask                  1.2.0
$ python -W default -m pytest tests
...\venv\lib\site-packages\pytest_flask\fixtures.py:6: DeprecationWarning: '_request_ctx_stack' is deprecated and will be removed in Flask 2.4.
  from flask import _request_ctx_stack
```

**Expected behavior**
Need to drop use of ``_request_ctx_stack`` to prepare for Flask 2.4 release.

**Environment (please complete the following information):**
 - OS: Windows 10
 - Python Version 3.10.0
 - pytest-flask version 1.2.0

**Comment by azmeuk** [comment]
Flask 3.0.0 and werkzeug 3.0.0 have been released yesterday. pytest-flask fails because of this now.

**Comment by kokobhara** [comment]
The`request_ctx` fixture already throws a warning that it's deprecated.

**Comment by vitalk** [comment]
Hi @ivanovmg

The latest released version (1.3.0) fixed the compatibility with `Flask 3.0`. Thank you for rising up this issue.
